src/CrudPanel.php

```diff
@@ -237,16 +237,61 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify a sub
+     * @param mixed $model Optionally specify a different model than the one in the crud object
      *
-     * @return String relation model name
+     * @return string relation model name
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode(".", $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
```

Comment:
Meaning what happens when an empty ```$relationString``` is passed? It will fail with a bad method call error. The relation string must define relation methods that exist in the models.